src/store/cockpitApi.ts

```diff
@@ -145,6 +146,79 @@ export const cockpitApi = emptyCockpitApi.injectEndpoints({
           }
         },
       }),
+      composeBlueprint: builder.mutation<
+        ComposeBlueprintApiResponse,
+        ComposeBlueprintApiArg
+      >({
+        queryFn: async ({ id: filename }) => {
+          const blueprintsDir = await getBlueprintsPath();
+          const file = cockpit.file(
+            path.join(blueprintsDir, filename, `${filename}.json`)
+          );
+          const contents = await file.read();
+          file.close();
+          const parsed = JSON.parse(contents);
+
+          const cloudapi = cockpit.http('/run/cloudapi/api.socket', {
```

Comment:
I have a PR to extract this to the basequery so we can get rid of this :)
And just use the basequery function.

Reply:
neat!